readthedocs/v3/serializers.py

```diff
@@ -0,0 +1,445 @@
+import datetime
+import urllib
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from rest_flex_fields import FlexFieldsModelSerializer
+from rest_flex_fields.serializers import FlexFieldsSerializerMixin
+from rest_framework import serializers
+
+from readthedocs.builds.models import Build, Version
+from readthedocs.projects.constants import LANGUAGES, PROGRAMMING_LANGUAGES
+from readthedocs.projects.models import Project
+
+
+class UserSerializer(FlexFieldsModelSerializer):
+
+    # TODO: return ``null`` when ``last_name`` or ``first_name`` are `''``. I'm
+    # thinking on writing a decorator or similar that dynamically creates the
+    # methods based on a field with a list
+
+    class Meta:
+        model = User
+        fields = [
+            'username',
+            'date_joined',
+            'last_login',
+            'first_name',
+            'last_name',
+        ]
+
+
+class BaseLinksSerializer(serializers.Serializer):
+
+    def _absolute_url(self, path):
+        scheme = 'http' if settings.DEBUG else 'https'
+        domain = settings.PRODUCTION_DOMAIN
+        return urllib.parse.urlunparse((scheme, domain, path, '', '', ''))
+
+
+class BuildTriggerSerializer(serializers.ModelSerializer):
```

Comment:
Feel like most of these classes could use docstrings to say what they are for.

Reply:
Yes. Writing docstrings for the whole PR is in my list before merging it.